server/core/storage.go

```diff
@@ -523,25 +523,49 @@ func (s *Storage) LoadMinServiceGCSafePoint(now time.Time) (*ServiceSafePoint, e
 	}
 	if len(keys) == 0 {
 		// There's no service safepoint. Store an initial value for GC worker.
-		return s.initServiceGCSafePointForGCWorker()
+		return s.initServiceGCSafePointForGCWorker(0)
 	}
 
-	min := &ServiceSafePoint{SafePoint: math.MaxUint64}
+	var min uint64 = math.MaxUint64
+	hasGCWorker := false
+	validMin := &ServiceSafePoint{SafePoint: math.MaxUint64}
 	for i, key := range keys {
 		ssp := &ServiceSafePoint{}
 		if err := json.Unmarshal([]byte(values[i]), ssp); err != nil {
 			return nil, err
 		}
+		if ssp.ServiceID == gcWorkerServiceSafePointID {
+			hasGCWorker = true
+			// If gc_worker's expire time is incorrectly set, fix it.
+			if ssp.ExpiredAt != math.MaxInt64 {
+				ssp.ExpiredAt = math.MaxInt64
+				err = s.SaveServiceGCSafePoint(ssp)
+				if err != nil {
+					return nil, errors.Trace(err)
+				}
+			}
+		}
+
+		if ssp.SafePoint < min {
```

Comment:
If ssp is expired, we may get a wrong `min` here. I suggest deleting `min` and keeping only `validMin`.